Lib/ntpath.py

```diff
@@ -679,11 +679,13 @@ def _getfinalpathname_nonstrict(path):
                     raise
                 try:
                     # The OS could not resolve this path fully, so we attempt
-                    # to follow the link ourselves. If we succeed, join the tail
-                    # and return.
-                    new_path = _readlink_deep(path)
-                    if new_path != path:
+                    # to follow the link ourselves. If we succeed, keep running the loop.
+                    new_path = _readlink_deep(path, seen)
+                    if new_path in seen:
                         return join(new_path, tail) if tail else new_path
```

Comment:
@eryksun's comment on the issue suggests `break`ing out the loop at this point (rather than `return`ing, presumably). From a cursory reading I think that would ensure the result doesn't include any symlinks, but would throw away the remaining parts of the input.

Reply:
Joining the tale and returning is correct. However, the code needs to normalize paths via `normcase()` before comparing or adding to `seen`.

Reply:
@eryksun Fixed in [1f8656e](https://github.com/python/cpython/pull/110751/commits/1f8656ec692b4c26e4f50c2638b3e74e8dcdbb2b). Can you have a look?

Reply:
`normcase()` returns a lowercase version of a path according to the OS. We only use it for tracking visited paths in `seen`, not for the result. You could compute it once as `new_path_norm = normcase(new_path)`.

Reply:
Yikes, fixed in [3ae56f0](https://github.com/python/cpython/pull/110751/commits/3ae56f073ba547ae2059d71f49917296368b817b).

Reply:
@eryksun Yikes, fixed in [3ae56f0](https://github.com/python/cpython/pull/110751/commits/3ae56f073ba547ae2059d71f49917296368b817b).